packages/dart/lib/src/utils/debug_logger.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:developer' as dev;
+
+import 'package:meta/meta.dart';
+
+import '../../sentry.dart';
+
+/// Lightweight isolate compatible diagnostic logger for the Sentry SDK.
+///
+/// Logger naming convention:
+/// - `sentry` – core dart package
+/// - `sentry.flutter` – flutter package
+/// - `sentry.{integration}` – integration packages (dio, hive, etc.)
+///
+/// Each package should have at least one top-level instance.
+///
+/// Example:
+/// ```dart
+/// const debugLogger = SentryDebugLogger('sentry.flutter');
+///
+/// debugLogger.warning('My Message')
+///```
+@internal
+class SentryDebugLogger {
+  final String _name;
+
+  const SentryDebugLogger(this._name);
+
+  static bool _isEnabled = false;
+  static SentryLevel _minLevel = SentryLevel.warning;
+
+  @visibleForTesting
+  static bool get isEnabled => _isEnabled;
+
+  @visibleForTesting
+  static SentryLevel get minLevel => _minLevel;
+
+  /// Configure logging for the current isolate.
+  ///
+  /// This needs to be called for each new spawned isolate before logging.
+  static void configure({
+    required bool isEnabled,
+    SentryLevel minLevel = SentryLevel.warning,
+  }) {
+    _isEnabled = isEnabled;
+    _minLevel = minLevel;
+  }
+
+  void debug(
+    String message, {
+    Object? error,
```

Comment:
Is there a case where we would log an error + stacktrace as debug? Same for `info` and probably also `warning`.